Source/Views/InfoLabel.swift

```diff
@@ -0,0 +1,145 @@
+import UIKit
+
+protocol InfoLabelDelegate: class {
+
+  func infoLabel(infoLabel: InfoLabel, didExpand expanded: Bool)
+}
+
+class InfoLabel: UILabel {
+
+  lazy var tapGestureRecognizer: UITapGestureRecognizer = {
+    let gesture = UITapGestureRecognizer()
+    gesture.addTarget(self, action: "labelDidTap:")
+
+    return gesture
+  }()
+
+  let model: LightboxModel
+  let numberOfVisibleLines = 2
+  weak var delegate: InfoLabelDelegate?
+  private var shortText = ""
+
+  var fullText: String {
+    didSet {
+      shortText = truncatedText
+      updateText(fullText)
+      configureLayout()
+    }
+  }
+
+  var ellipsis: String {
+    return "... \(model.infoLabel.ellipsisText)"
+  }
+
+  var expandable: Bool {
+    return shortText != fullText
+  }
+
+  private(set) var expanded = false {
+    didSet {
+      delegate?.infoLabel(self, didExpand: expanded)
+    }
+  }
+
+  private var truncatedText: String {
+    var truncatedText = fullText
+
+    guard numberOfLines(fullText) > numberOfVisibleLines else {
+      return truncatedText
+    }
+
+    truncatedText += ellipsis
+
+    var range = Range<String.Index>(
+      start: truncatedText.endIndex.advancedBy(-(ellipsis.characters.count + 1)),
+      end: truncatedText.endIndex.advancedBy(-ellipsis.characters.count)
+    )
+
+    while numberOfLines(truncatedText) > numberOfVisibleLines {
+      truncatedText.removeRange(range)
+      range.startIndex = range.startIndex.advancedBy(-1)
+      range.endIndex = range.endIndex.advancedBy(-1)
+    }
+
+    return truncatedText
+  }
+
+  // MARK: - Initialization
+
+  init(model: LightboxModel, text: String, expanded: Bool = false) {
+    self.model = model
+    self.fullText = text
+    super.init(frame: CGRectZero)
+
+    numberOfLines = 0
```

Comment:
You could skip `self` in `init` as well unless you have attribute with the same name.